site/cypress/e2e/Cypress-Feature/test_forum.spec.js

```diff
@@ -23,16 +23,44 @@ const createThread = (title, content, category) => {
 };
 
 const replyToThread = (title, reply) => {
-    cy.get('.thread-left-cont > .thread-list-item').contains(title).click();
-    cy.get('.create-post-head').should('contain', title);
+    cy.get('[data-testid="thread-list-item"]').contains(title).click();
+    cy.get('[data-testid="create-post-head"]').should('contain', title);
     cy.get('#reply_box_3').type(reply);
     cy.get('[value="Submit Reply to All"]').should('not.be.disabled').click();
     cy.get('#posts_list').should('contain', reply);
 };
 
+const upduckPost = (thread_title) => {
+    cy.get('[data-testid="thread-list-item"]').contains(thread_title).click();
+    cy.get('[data-testid="create-post-head"]').should('contain', thread_title);
+    cy.get('[data-testid="like-count"]').first().should('have.text', 0);
+    cy.get('[data-testid="upduck-button"]').first().click();
+    cy.get('[data-testid="like-count"]').first().should('have.text', 1);
+};
+
+const checkStatsUpducks = (fullName, numUpducks) => {
+    // Check the stats page for a user with fullName and
+    // number of upducks numUpducks
+    cy.get('[data-testid="more-dropdown"]').click();
+    cy.get('#forum_stats').click();
+    let found = false;
+    cy.get('[data-testid="user-stat"]').each(($el) => {
+        const name = $el.find('td').first().text().trim();
+        if (name === fullName) {
+            found = true;
+            const upducks = parseInt($el.find('[data-testid="upduck-stat"]').text().trim());
+            expect(upducks).to.eq(numUpducks);
+            return false;
+        }
+    }).then(() => {
+        expect(found).to.be.true;
+        cy.get('[title="Back to threads"]').click();
+    });
```

Comment:
I believe this can all be trimmed down to 
```suggestion
    cy.get('[data-testid="user-stat"]').contains(fullName).siblings('[data-testid="upduck-stat"]').should('contain.text', numUpducks);
    cy.get('[title="Back to threads"]').click();
```